airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,475 @@
+
+        session.commit()
+        self.driver = driver
+        Session.remove()
+
+    def status_update(self, update):
+        """Update the Status of the Tasks. Based by Mesos Events."""
+        self.log.info(
+            "Task %s is in state %s", update['status']['task_id']['value'], update['status']['state'])
+
+        try:
+            key = self.task_key_map[update['status']['task_id']['value']]
+
+        except KeyError:
+            # The map may not contain an item if the framework re-registered
+            # after a failover.
+            # Discard these tasks.
+            self.log.info("Unrecognised task key %s", update['status']['task_id']['value'])
+            return
+
+        if update['status']['state'] == "TASK_RUNNING":
+            self.result_queue.put((key, State.RUNNING))
+            return
+
+        if update['status']['state'] == "TASK_FINISHED":
+            self.result_queue.put((key, State.SUCCESS))
+            return
+
+        if update['status']['state'] == "TASK_LOST" or \
+           update['status']['state'] == "TASK_KILLED" or \
+           update['status']['state'] == "TASK_FAILED":
+            self.result_queue.put((key, State.FAILED))
+            return
+
+
+class MesosExecutor(BaseExecutor):
+    """
+    MesosExecutor allows distributing the execution of task
+    instances to multiple mesos workers.
+
+    Apache Mesos is a distributed systems kernel which abstracts
+    CPU, memory, storage, and other compute resources away from
+    machines (physical or virtual), enabling fault-tolerant and
+    elastic distributed systems to easily be built and run effectively.
+    See http://mesos.apache.org/
+    """
+
+    class MesosFramework(threading.Thread):
+        """MesosFramework class to start the threading"""
+
+        def __init__(self, client):
+            threading.Thread.__init__(self)
+            self.client = client
+            self.stop = False
+
+        def run(self):
+            try:
+                self.client.register()
+            except KeyboardInterrupt:
+                print('Stop requested by user, stopping framework....')
+
+    def __init__(self):
+        super().__init__()
+        self.commands_to_run = []
+        self.task_queue = Queue()
+        self.result_queue = Queue()
+        self.driver = None
+        self.client = None
+        self.th = None  # pylint: disable=invalid-name
+
+    def start(self):
+        """Setup and start routine to connect with the mesos master"""
+        if not conf.get('mesos', 'MASTER'):
+            self.log.error("Expecting mesos master URL for mesos executor")
+            raise AirflowException("mesos.master not provided for mesos executor")
+
+        master = conf.get('mesos', 'MASTER')
+
+        framework_name = get_framework_name()
+        framework_id = None
+
+        if not conf.get('mesos', 'TASK_CPU'):
+            task_cpu = 1
+        else:
+            task_cpu = conf.getint('mesos', 'TASK_CPU')
```

Comment:
```suggestion
        task_cpu  = conf.get('mesos', 'TASK_CPU', fallback=1):
```
This should work and will reduce number of code (here and in other places)

Reply:
hmmm... thats great. :-) thanks